libpod/pod.go

```diff
@@ -97,6 +98,8 @@ type InfraContainerConfig struct {
 	HasInfraContainer  bool                  `json:"makeInfraContainer"`
 	NoNetwork          bool                  `json:"noNetwork,omitempty"`
 	HostNetwork        bool                  `json:"infraHostNetwork,omitempty"`
+	Pid                specgen.Namespace     `json:"infraPid,omitempty"`
+	LinuxPidNS         specs.LinuxNamespace  `json:"LinuxPIDNS,omitempty"`
```

Comment:
Do we need both of these? I would imagine we can get the mode just by looking at the `specs.LinuxNamespace` - or, alternatively, generate the `specs.LinuxNamespace` from the `specgen.Namespace` when creating the infra container?

Reply:
@mheon I kept both of them in order to parse the Pid NS before runtime. The types are different as `specgen.Namespace` contains info like `host` or `pod` which I do use again at runtime but `specs.LinuxNamespace` is the actual NS struct containing `pid` and the proper path. Not sure if I can parse before runtime and not have both of these but I will look.

Reply:
fixed